### ISSUE TITLE
ScatterplotLayer({data: {async iterable}}) may place circles at incorrect positions

### ISSUE THREAD

**Issue by kkdd** [uber/deck.gl#3838]
Hello,

#### Description
```ScatterplotLayer({data: {async iterable}})``` may seem to place circles at incorrect positions.

The following is the example that plots circles near incorrect position {longitude:0, latitude:0}:

![scatter](https://user-images.githubusercontent.com/5372642/68089883-c5ef3c00-feb0-11e9-8cae-ce1357af74b4.jpg)

#### Repro Steps

```
npx create-react-app scatter
cd scatter
vi package.json
vi src/App.js
npm install
npm run build
npm start
```

```cat src/App.js```:

```
import React, {useState} from "react";
import {StaticMap} from 'react-map-gl';
import DeckGL, {ScatterplotLayer} from "deck.gl";

const MAPBOX_TOKEN = process.env.REACT_APP_MapboxAccessToken;
// echo REACT_APP_MapboxAccessToken=xxxxxxxxx > .env

export default () => {
    const maxIter = 8000;
    const delayTime = 40; // in msec
    const defaultColor = [255,64,255];

    function _delay(msec) {
        return new Promise(resolve => setTimeout(resolve, msec));
    }
  
    function _dat(i) {
        var a = i*5;
        return {coordinates: [a%360-180, a%35+15]};
    }
    
    async function* _getData() {
        for (let i = 0; i < maxIter; i++) {
            await _delay(delayTime);
            yield _dat(i);
        }
    }

    function _renderLayers(props) {
        const { data } = props;
        const scatter = new ScatterplotLayer({
            id: "scatter-layer",
            data: data,
            radiusMinPixels: 3,
            radiusMaxPixels: 4,
            getPosition: d => d.coordinates,
            getFillColor: d => ('color' in d ? d.color : defaultColor),
        })
        const layes = [scatter];
        return layes;
    };

    const [viewport] = useState({
        zoom: 0,
        longitude: 0,
        latitude: 0,
        width: window.innerWidth,
        height: window.innerHeight,
    });

    return (
        <div style={{ margin: '0 auto'}}>
        <DeckGL
          initialViewState={viewport}
          layers={_renderLayers({data: _getData()})}
          controller={true}
        >
        <StaticMap mapboxApiAccessToken={MAPBOX_TOKEN} mapStyle="mapbox://styles/mapbox/light-v9" />
        </DeckGL>
        </div>
    );
};
```

```cat package.json```:

```
{
  "name": "scatter",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "react": "^16.11.0",
    "react-dom": "^16.11.0",
    "react-scripts": "3.2.0",
    "deck.gl": "7.3.3",
    "react-map-gl": "5.1.1"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

#### Environment (please complete the following information):
 - Framework Version: "deck.gl": "7.3.3"
 - Browser Version: Chrome 78.0.3904.87
 - OS: Mac OS X 10.13.6（17G9016）

**Comment by Pessimistress** [comment]
Async iterator is expected to yield arrays of objects.

**Comment by kkdd** [comment]
Thank you for your comment.
Is this a bug originate from myself? Could you give me some additional suggestions?

**Comment by Pessimistress** [comment]
Try replace `yield _dat(i);` with `yield [_dat(i)];`

**Comment by kkdd** [comment]
Thank you for your suggestion.
I regret to say that the replace mentioned results in producing the same output as before.

**Comment by Pessimistress** [comment]
@kkdd There is indeed a bug when we resize buffers. I will put up a fix shortly.

**Comment by kkdd** [comment]
Thank you.

**Comment by Pessimistress** [comment]
Published 7.3.5.
